### PR TITLE
Browse Plugins v1

### PR DESCRIPTION
**Pull request by oguzkocer**
This is the initial version of browse plugin feature. The network call and it's payloads are mostly figured out right now, but before going through the polishing process I wanted to open the PR as is to keep things small. I'll also add some integration tests later on while I polish things out.

/cc @tonyr59h

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/PluginStore.java [hunk 1]
```diff
@@ -148,7 +162,7 @@ private void fetchPluginInfo(String plugin) {
         mPluginWPOrgClient.fetchPluginInfo(plugin);
     }
 
-    private void fetchPluginDirectory(PluginWPOrgClient.FetchPluginDirectoryPayload payload) {
+    private void fetchPluginDirectory(FetchPluginDirectoryPayload payload) {
```
Comment (tonyr59h) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/PluginStore.java]: 👍

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/plugin/PluginRestClient.java [hunk 2]
```diff
@@ -37,7 +37,7 @@
 @Singleton
 public class PluginRestClient extends BaseWPComRestClient {
     // Payloads
-    public static class UpdateSitePluginPayload extends Payload {
+    public static class UpdateSitePluginPayload extends Payload<BaseNetworkError> {
```
[the remark this answers is not in the document]
Comment (oguzkocer) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/plugin/PluginRestClient.java]: Probably not? This payload won't use the error, so it's kind of weird to even have an error type for it.

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/wporg/plugin/PluginWPOrgClient.java [hunk 3]
```diff
@@ -43,13 +47,36 @@
         }
     }
 
-    public static class FetchPluginDirectoryPayload extends Payload {
+    public static class FetchPluginDirectoryPayload extends Payload<BaseNetworkError> {
```
[the remark this answers is not in the document]
Comment (oguzkocer) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/wporg/plugin/PluginWPOrgClient.java]: Similar to [this one](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/568#discussion_r141001971), I don't think so.

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/wporg/plugin/PluginWPOrgClient.java [hunk 4]
```diff
@@ -43,13 +47,36 @@
...
         public int page;
         public int pageSize;
+        public PluginDirectoryType directoryType;
 
         public FetchPluginDirectoryPayload() {
             page = 1;
             pageSize = 30;
+            directoryType = PluginDirectoryType.POPULAR;
```
[the remark this answers is not in the document]
Comment (oguzkocer) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/wporg/plugin/PluginWPOrgClient.java]: This is temporary and should be fixed in a later PR as I don't know what we should default to.

**Comment by oguzkocer** [comment]
@tonyr59h I think this is ready for another pass. As mentioned, some changes are coming in a later PR™.

**Comment by tonyr59h** [comment]
:shipit:

**Pull request merged**